buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersTask.java

```diff
@@ -0,0 +1,14 @@
+package org.elasticsearch.gradle.testclusters;
+
+import org.gradle.api.Task;
+import org.gradle.api.tasks.Nested;
+
+import java.util.Collection;
+
+public interface TestClustersTask extends Task {
```

Comment:
There's really no reason for this to extend `Task` as there's no reason why things that aren't tasks can't use a test cluster as an input (think some other `@Nested`) thing.

Secondly, given that, I'd rename this to `TestClustersAware`.

Lastly, now that interfaces can contain instance variables, I'd make this a fully functional interface with default implementations for both these methods. That way any task can become "TestClustersAware" by virtue of simply implementing this interface.

Reply:
I did the rename, but instance variables won't work as they are static, so we'll have to live with the duplication for now.

Reply:
Also, extending task is necessary for `tasks.withType` that we rely on in a few places.

Reply:
> but instance variables won't work as they are static
Dammit, right. 👍